packages/app/src/providers/mock/mock-api.ts

```diff
@@ -75,16 +75,14 @@ export class MockApi extends DataProviderApi {
       await this.cancellableDelay(params);
     }
 
-    assertMockDataset(dataset);
-    const { value: rawValue } = dataset;
-    const value = this.processRawValue(dataset, rawValue);
+    const { value } = dataset;
+    assertDefined(value, 'Expected mock dataset to have value');
```

Comment:
We don't need to process the value anymore, since we already store flat arrays in the mock metadata (cf. the `array()` metadata utility function).